libraries/ts/src/margin/marginAccount.ts

```diff
@@ -375,16 +375,15 @@ export class MarginAccount {
       }
     }
 
+    // Wallet's balance for pool
+    // If depsiting or repaying SOL, maximum input should consider fees
     let walletAmount = (pool.symbol && this.walletTokens?.map[pool.symbol].amount) ?? TokenAmount.zero(pool.decimals)
-
-    // If depsiting or repaying SOL, maximum input should still cover fees
     if (pool.tokenMint.equals(NATIVE_MINT)) {
-      const feeCover = TokenAmount.tokens(0.07, pool.decimals)
-      walletAmount = TokenAmount.max(walletAmount.sub(feeCover), TokenAmount.zero(pool.decimals))
+      walletAmount = TokenAmount.max(walletAmount.subn(feesBuffer), TokenAmount.zero(pool.decimals))
```

Comment:
`subn` `muln` and all the other number functions are very unreliable when numbers get big
```suggestion
      walletAmount = TokenAmount.max(walletAmount.sub(numberToBN(feesBuffer)), TokenAmount.zero(pool.decimals))
```